projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/asm/LLVMAMD64AdcNode.java

```diff
@@ -53,8 +93,21 @@ public LLVMAMD64AdcbNode(LLVMAMD64UpdateCPZSOFlagsNode flags) {
         protected byte executeI8(VirtualFrame frame, byte left, byte right, boolean cf) {
             byte c = (byte) (cf ? 1 : 0);
             byte result = (byte) (left + right + c);
-            boolean overflow = (result < 0 && left > 0 && right > 0) || (result > 0 && left < 0 && right < 0);
-            boolean carry = ((left < 0 || right < 0) && result > 0) || (left < 0 && right < 0);
+            boolean overflow;
+            boolean carry;
+            if (!cf) {
```

Comment:
We should profile these conditions (here and in the other execute methods of this class).